osler/templates/inventory/inventory-main.html

```diff
@@ -35,25 +35,22 @@ <h2>Drug Inventory</h2>
                 <td><b>{{ drug.category }}</b></td>
                 <td><b>{{ drug.stock }}</b></td>
                 <td>{{ drug.lot_number }}</td>
-                <td>{% if drug.dose %}
-                        {{ drug.dose }}
-                    {% endif %}
-                    {% if drug.unit %}
-                        {{ drug.unit }}
-                    {% endif %}</td>
-                <td>{% if drug.expiration_date %}
-                        {{ drug.expiration_date }}
-                    {% endif %}</td>
-                <td>{% if drug.manufacturer %}
-                        {{ drug.manufacturer }}
-                    {% endif %}
+                <td>{{ drug.dose }}{{ drug.unit }}</td>
+                <td>{% if drug.expired %}
+                  <b style="color:red;">{{ drug.expiration_date }}</b>
+                  {% elif drug.pre_expire %}
+                  <b style="color:gold;">{{ drug.expiration_date }}</b>
+                  {% else %}
+                  {{ drug.expiration_date }}
+                  {% endif %}
                 </td>
+                <td>{{ drug.manufacturer }}</td>
+
                 <td>
                     <a href="#" class="btn btn-success a-btn-slide-text btn-xs" data-toggle="modal" data-target="#dispenseModal{{ drug.id }}">
                         <span class="glyphicon glyphicon-minus" aria-hidden="true"></span>
                         <span><strong>Dispense</strong></span>
                     </a>
-                    <!-- Modal -->
                     <div class="modal fade" id="dispenseModal{{ drug.id }}" tabindex="-1" role="dialog" aria-labelledby="dispenseModal{{ drug.id }}Label" aria-hidden="true">
                     <form action="{% url 'inventory:drug-dispense' %}" method="post">
```

Comment:
Ok so here you are basically manually making the html for a form in the modal. Now that it's more complex than it used to be, like Justin suggested back in the views.py, just put a form here. Define a form in forms.py like DispenseForm. Then you won't have to manually define all these input types and such and you can clean up your dispense_drug view.

Reply:
Sounds good! Thanks for looking over my code.